kod/object/active/holder/room/monsroom/g9.kod

```diff
@@ -256,317 +256,198 @@ messages:
+         #fine_row=17,#fine_col=32,#new_angle=ANGLE_SOUTH);
       
-      propagate;
-   }
 
-   DisableIllusions()
-   {
-      local oPortal;
-      if not (Send(self,@CouNtholdingHowMany,#class=&portal) > 0)
-      {
-         oPortal = Create(&Portal,#dest_room_num=RID_G9,#dest_row=7,#dest_col=55,
-                          #dest_fine_row=39,#dest_fine_col=60);
-         Send(self,@NewHold,#what=oPortal,#new_row=20,#new_col=56,
-              #fine_row=0,#fine_col=0,#new_angle=ANGLE_SOUTH_EAST);
-      }
-
-      Send(self,@SetSector,#sector=2,#animation=ANIMATE_FLOOR_LIFT,
-           #height=224,#speed=0);
-      Send(self,@SetSector,#sector=3,#animation=ANIMATE_FLOOR_LIFT,
-           #height=224,#speed=0);
-
-      return;
-   }
-
-   EnableIllusions()
-   {
-      local i, each_obj;
-
-      for i in plActive
-      {
-         each_obj = Send(self,@HolderExtractObject,#Data=i);
-         if IsClass(each_obj,&portal)
-         {
-            Send(each_obj,@Delete);
-         }
-      }
+      % Create platform levers
+      % Note: The platform levers are the current solution to the puzzle.
 
-      Send(self,@SetSector,#sector=2,#animation=ANIMATE_FLOOR_LIFT,
-           #height=400,#speed=0);
-      Send(self,@SetSector,#sector=3,#animation=ANIMATE_FLOOR_LIFT,
-           #height=264,#speed=0);
-
-      return;
-   }
+      poCenterLever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poCenterLever,#new_row=21,#new_col=55,#fine_row=30,#fine_col=30);
+      
+      poNWLever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poNWLever,#new_row=16,#new_col=51,#fine_row=0,#fine_col=0);
+      
+      poNELever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poNELever,#new_row=16,#new_col=60,#fine_row=0,#fine_col=0);
 
-   FirstUserEntered()
-   {
-      if NOT (Send(SYS,@AreRoomIllusionsEnabled))
-      {
-         Send(self,@SetSector,#sector=2,#animation=ANIMATE_FLOOR_LIFT,
-              #height=224,#speed=0);
-         Send(self,@SetSector,#sector=3,#animation=ANIMATE_FLOOR_LIFT,
-              #height=224,#speed=0);
-      }
+      poSWLever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poSWLever,#new_row=25,#new_col=51,#fine_row=0,#fine_col=0);
+      
+      poSELever = Create(&Lever,#description=G9_floor_lever_desc,#range=1);
+      Send(self,@NewHold,#what=poSELever,#new_row=25,#new_col=60,#fine_row=0,#fine_col=0);
 
       propagate;
    }
-
-   NewHold(what=$, new_row=$, new_col=$)
+   
+   ReturnPuzzleToInitialState()
+   "Returns levers, lights, and spire to their original state."
    {
-      if NOT (Send(SYS,@AreRoomIllusionsEnabled))
-         OR ((what=$) OR (new_row=$) OR (new_col=$))
-         OR Send(poPlatform,@GetState) = SECTOR_UP
-      {
-         propagate;
-      }
+      ptResetPuzzle = $;
```

Comment:
Possible bug: need to DeleteTimer on ptResetPuzzle?

Reply:
I _think_ this is good. In my latest commit(s), `ptResetPuzzle` only stores timers that callback to this function -- `ReturnPuzzleToInitialState`. But maybe I'm not thinking about it clearly?

Reply:
So if someone accidentally calls this from admin mode, because from the name it looks like it resets the puzzle, what happens?